test_conformance/images/kernel_read_write/test_common.cpp

```diff
@@ -34,122 +34,210 @@ cl_sampler create_sampler(cl_context context, image_sampler_data *sdata, bool te
     return sampler;
 }
 
-void InitFloatCoordsCommon(image_descriptor *imageInfo,
-                           image_sampler_data *imageSampler, float *xOffsets,
-                           float *yOffsets, float *zOffsets, float xfract,
-                           float yfract, float zfract, int normalized_coords,
-                           MTdata d, int lod)
+cl_int get_image_dimensions(image_descriptor *imageInfo, size_t &width,
```

Comment:
If we return _cl_int_, the return type should be changed to _test_status_ (or you could return a _bool_, or define your own type, or ...).

Reply:
I've gone for bool, as test_status feels like we should return TEST_PASS if successful, which isn't true at this point.

Reply:
Looks good to me.

I did notice it returns either TEST_FAIL or 0 which get implicitly converted to bools, which isn't ideal, but is not worth holding the change up for.  I agree about fixing these small details later.